README.md

```diff
@@ -120,8 +151,39 @@ From this stage onward your device will begin training. You should see your peer
 You can also track your training progress in real time:
 - On the Gensyn Testnet Dashboard: [dashboard.gensyn.ai](https://dashboard.gensyn.ai)
 
+## Environment Overview (CodeZero)
+
+CodeZero extends RL Swarm with distinct node roles that collaborate on programming challenges:
+
+| Role | Description | Typical Host |
+|------|--------------|---------------|
+| **Solver** | Learns locally on code tasks using GRPO; shares rollouts with peers | Community node |
+| **Proposer** | Generates coding problems and adjusts difficulty heuristically | Network service |
+| **Evaluator** | Frozen model predicting correctness and assigning rewards | Hosted service |
```

Comment:
Can we remove the typical host column? We want to avoid explicitly calling out the centralized pieces of this.